### PR TITLE
Stretching

### PR DESCRIPTION
**Pull request by AaronWoodward**
This PR adds a stretching tool (for generic models only) which acts like inflate, but is multiplicative and can be done on each axis individually (or on all of them in unison). It can also be disabled in the settings.

As suggested, I went through all mentions of `inflate` and updated accordingly to account for stretching.

The only instance I wasn't sure how to handle is `validator.js` line 170. @JannisX11 
I also wasn't sure what the files in the second commit are about - so I separated them for easy removal if they aren't meant to be merged.

https://user-images.githubusercontent.com/20425809/229634036-cdffc1e6-be5d-4755-bf89-66ff0c408099.mp4

### CODE REVIEW

File: js/interface/actions.js [hunk 1]
```diff
@@ -2219,6 +2219,18 @@ const BARS = {
 				'slider_inflate'
 			]
 		})
+		Toolbars.element_stretch = new Toolbar({
+			id: 'element_stretch',
+			name: 'panel.element.stretch',
+			label: true,
+			condition: () => settings.stretch_enabled.value && Format.id === "free",
```
Comment (JannisX11) [inline js/interface/actions.js:2226]: Let's use a format feature flag to condition the stretch feature.
The flag can be defined here https://github.com/JannisX11/blockbench/blob/master/js/io/format.js#L241 like the others and can be accessed via:
`Format.stretch_cubes`
or
`{features: ['stretch_cubes']}`
The flag can be enabled when defining a format. This way we can re-use this if we ever create a dedicated format, or other people want to use this feature in their formats.

File: js/modeling/transform.js [hunk 2]
```diff
@@ -1194,6 +1194,109 @@ BARS.defineActions(function() {
 		}
 	})
 
+	//Stretch
+	new NumSlider('slider_stretch_x', {
+		name: tl('action.slider_stretch', ['X']),
+		description: tl('action.slider_stretch.desc', ['X']),
+		color: 'x',
+		category: 'transform',
+		condition: function() {return Cube.selected.length && Modes.edit},
```
Comment (JannisX11) [inline js/modeling/transform.js:1203]: The sliders and link toggle should also be conditioned on the feature flag. Yeah, the toolbar gets disabled, but users would still be able to add these to other toolbars and they wouldn't work correctly.

File: js/modeling/transform.js [hunk 3]
```diff
@@ -1643,6 +1746,16 @@ BARS.defineActions(function() {
 			}
 		}
 	})
+	new Toggle('toggle_stretch_linked', {
+		icon: 'fas.fa-link',
+		category: 'transform',
+		linked_setting: 'stretch_linked',
+		condition: {
+			modes: ['edit', 'animate'],
```
Comment (JannisX11) [inline js/modeling/transform.js:1754]: The toggle should ideally have the same conditions as the sliders. So enabled only when cubes are enabled, not any type of element. And only enabled in edit mode, not in animate mode.

File: service_worker.js
Comment (JannisX11) [inline service_worker.js:1]: No need to include these auto-generated files like package-lock and service_worker in the PR. I only build them before each update.

File: js/outliner/cube.js [hunk 4]
```diff
@@ -863,6 +877,14 @@ new Property(Cube, 'boolean', 'locked');
 
 OutlinerElement.registerType(Cube, 'cube');
 
+function adjustFromAndToForInflateAndStretch(from, to, element) {
+	for (let i = 0; i < from.length; i++) {
+		from[i] = (from[i] - element.inflate) * element.stretch[i];
```
Comment (JannisX11) [inline js/outliner/cube.js:882]: Multiplying the absolute coordinates of the cube currently not only stretches the cube, but also moves it's position away from the scene origin. The stretch should be calculated from the local center of the cube (the middle value between from and to).

**Review by JannisX11 — CHANGES_REQUESTED** [review]
Here's my review. A couple changes are needed to integrate this.

The integration of stretch in validator.js L170 would be something like this:
```javascript
let invalid_size_axes = size.filter((value, axis) => value < 0.999 && (value+cube.inflate*2) * cube.stretch[axis] > 0.005);
```

**Comment by AaronWoodward** [comment]
@JannisX11 Thanks for the great feedback! I believe I have addressed it all.

**Pull request merged**